apm-agent-plugin-sdk/src/main/java/co/elastic/apm/agent/sdk/ElasticApmInstrumentation.java

```diff
@@ -89,6 +89,17 @@
  *
  */
 public abstract class ElasticApmInstrumentation {
+
+    private boolean used;
```

Comment:
Instead of making the instrumentations aware of whether they have been used, I'd encapsulate this state in a dedicated class that holds a collection of used/unused instrumentations or just two sets of Strings for used/unused.

Reply:
Done